userspace/libsinsp/container_info.h

```diff
@@ -242,7 +256,7 @@ class sinsp_container_info
 		std::vector<std::string> m_health_probe_args;
 	};
 
-	sinsp_container_info():
+	sinsp_container_info(sinsp_container_lookup &&lookup = sinsp_container_lookup()):
```

Comment:
This  goes beyond my bread-and-butter knowledge of move semantics... What's the use case? Only rvalue references are acceptable as a parameter? If not, wouldn't a simpler pass-by-value work? If it is to stay this way for a reason, do we need the `std::move` below? Wouldn't be moved anyway since it's an rvalue reference?

Reply:
The only valid usage we have for now is with an rvalue, let's keep it this way for now.

Reply:
Surprisingly enough, without the `std::move` it will try to call the copy ctor: https://wandbox.org/permlink/6jPkpNHH5lmeqXyx

From https://en.cppreference.com/w/cpp/utility/move:
> Names of [rvalue reference](https://en.cppreference.com/w/cpp/language/reference) variables are [lvalues](https://en.cppreference.com/w/cpp/language/value_category) and have to be converted to [xvalues](https://en.cppreference.com/w/cpp/language/value_category) to be bound to the function overloads that accept rvalue reference parameters, which is why [move constructors](https://en.cppreference.com/w/cpp/language/move_constructor) and [move assignment operators](https://en.cppreference.com/w/cpp/language/move_operator) typically use std::move